File: Greedy/ActivitySelection/select.py
class Solution:
    def activitySelection(self, start: list[int], finish: list[int]) -> int:
        #code here
        n = len(start)
        activities = sorted(zip(start, finish), key=lambda x: x[1])
        currEnd = activities[0][1]
        removals = 0
        for i in range(1,n):
            nextStart = activities[i][0]
            nextEnd = activities[i][1]
            if nextStart <= currEnd:
                removals += 1
                currEnd = min(currEnd,nextEnd)
            else:
                currEnd = nextEnd
        totalActivities = n-removals
        return totalActivities

File: Greedy/ActivitySelection/select.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    static bool Compare(const pair<int, int>& a, const pair<int, int>& b) {
        return a.second < b.second;
    }

    int activitySelection(vector<int>& start, vector<int>& finish) {
        int n = start.size();

        vector<pair<int, int>> activities;

        for (int i = 0; i < n; i++) {
            activities.push_back({start[i], finish[i]});
        }

        sort(activities.begin(), activities.end(), Compare);

        int currEnd = activities[0].second;
        int removals = 0;

        for (int i = 1; i < n; i++) {
            int nextStart = activities[i].first;
            int nextEnd = activities[i].second;

            if (nextStart <= currEnd) {
                removals++;
                currEnd = min(currEnd, nextEnd);
            } else {
                currEnd = nextEnd;
            }
        }

        return n - removals;
    }
};